Revit_Core_Engine/Query/DuctShape.cs

```diff
@@ -54,15 +47,28 @@ public static Autodesk.Revit.DB.ConnectorProfileType DuctShape(this Autodesk.Rev
                 return Autodesk.Revit.DB.ConnectorProfileType.Invalid;
             }
 
-            string ductFamilyName = duct?.DuctType?.FamilyName;
+            ConnectorProfileType connectorShape = ConnectorProfileType.Invalid;
 
+#if REVIT2018
+            // Get the duct connector shape in Revit 2018 by extracting it from one of the duct connectors
+            var connectors = duct.ConnectorManager.Connectors;
```

Comment:
I believe this needs to take the first `End` type connector instead of any connector.

Reply:
Revised to find the right connector